scripts/error-extractor/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+var path = require('path');
+var traverse = require('babel-traverse').default;
+
+var evalToString = require('./evalToString');
+var getFilePaths = require('./getFilePaths');
+var invertObject = require('./invertObject');
+var parseJSFile = require('./parseJSFile');
+var writeJSON = require('./writeJSON');
+
+var outputFileName = 'ErrorMap.json';
+var outputFilePath = path.join(__dirname, '../../', outputFileName);
+
+var existingErrorMap;
+var currentID;
+
+try {
```

Comment:
Instead of having this code at the top level, can you extract this into functions and then just call the entry point at the end of the file? That makes it easier to reuse the code later if we want to use this logic for something else. The I/O can also probably be in a different function than the logic to populate the map.